contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -488,6 +488,11 @@ static VALUE rb_trilogy_connect(VALUE self, VALUE encoding, VALUE charset, VALUE
             Check_Type(val, T_FIXNUM);
             connopt.port = NUM2USHORT(val);
         }
+
+        if ((val = rb_hash_lookup(opts, ID2SYM(id_ip_address))) != Qnil) {
+            Check_Type(val, T_STRING);
+            connopt.ip_address = StringValueCStr(val);
```

Comment:
Do we need to free `ip_address` as part of `_cb_raw_close`?

Reply:
Good catch, I added it.